File: nodal_service1/demo/src/main/java/com/nodalexchange/demo/RabbitMQConstants.java
package com.nodalexchange.demo;

public final class RabbitMQConstants {

    // Exchange name
    public static final String EXCHANGE_NAME = "nodalExchange";

    // Queue names
    public static final String QUEUE_NAME_SERVICE1 = "nodalService1Queue";
    public static final String QUEUE_NAME_SERVICE2 = "nodalService2Queue";

    // Routing keys
    public static final String ROUTING_KEY_SERVICE1 = "service1";
    public static final String ROUTING_KEY_SERVICE2 = "service2";

    // Messages
    public static final String MESSAGE_PING = "ping";
    public static final String MESSAGE_PONG = "pong";

    // Private constructor to prevent instantiation
    private RabbitMQConstants() {
        throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
    }
}

File: nodal_service2/demo/src/main/kotlin/com/nodalexchange/demo/RabbitMQConstants.kt
package com.nodalexchange.demo

object RabbitMQConstants {
    // Exchange name
    const val EXCHANGE_NAME = "nodalExchange"

    // Queue names
    const val QUEUE_NAME_SERVICE1 = "nodalService1Queue"
    const val QUEUE_NAME_SERVICE2 = "nodalService2Queue"

    // Routing keys
    const val ROUTING_KEY_SERVICE1 = "service1"
    const val ROUTING_KEY_SERVICE2 = "service2"

    // Messages
    const val MESSAGE_PING = "ping"
    const val MESSAGE_PONG = "pong"
}
